src/Consolonia.Core/Drawing/PixelBufferImplementation/Pixel.cs

```diff
@@ -102,56 +102,43 @@ public bool Equals(Pixel other)
         public Pixel Blend(Pixel pixelAbove)
         {
             PixelForeground newForeground;
-            PixelBackground newBackground;
 
-            if (pixelAbove.IsCaret) return new Pixel(Foreground, Background, true);
+            var newBackground = new PixelBackground(MergeColors(Background.Color, pixelAbove.Background.Color));
 
-            switch (pixelAbove.Background.Mode)
+            bool newIsCaret;
+
+            if (pixelAbove.Background.Color.A == 0x0 /*todo: can be approximate, extract to extension method*/)
+            {
+                newForeground = pixelAbove.Foreground.NothingToDraw()
+                    ? Foreground
+                    : Foreground.Blend(pixelAbove.Foreground);
+                newIsCaret = pixelAbove.IsCaret | IsCaret;
+            }
+            else
             {
-                case PixelBackgroundMode.Colored:
-                    // merge pixelAbove into this pixel using alpha channel.
-                    Color mergedColors = MergeColors(Background.Color, pixelAbove.Background.Color);
+                if (!pixelAbove.Foreground.NothingToDraw())
+                {
                     newForeground = pixelAbove.Foreground;
-                    newBackground = new PixelBackground(mergedColors);
-                    return new Pixel(newForeground, newBackground, pixelAbove.IsCaret);
-
-                case PixelBackgroundMode.Transparent:
-                    // if the foreground is transparent, ignore pixelAbove foreground.
-                    newForeground = pixelAbove.Foreground.Color != Colors.Transparent
-                        ? Foreground.Blend(pixelAbove.Foreground)
-                        : Foreground;
-
-                    // background is transparent, ignore pixelAbove background.
-                    newBackground = Background;
-                    break;
-                case PixelBackgroundMode.Shaded:
-                    // shade the current pixel
-                    (newForeground, newBackground) = Shade();
-
-                    // blend the pixelAbove foreground into the shaded pixel
-                    newForeground = newForeground.Blend(pixelAbove.Foreground);
-
-                    // resulting in new pixel with shaded background and blended foreground
-                    return new Pixel(newForeground, newBackground);
-
-                default: throw new ArgumentOutOfRangeException(nameof(pixelAbove));
+                }
+                else
+                {
```

Comment:
add a comment to make it clear the rest of the arguments are passed through
// merge the PixelForeground color with the pixelAbove backbround color

Reply:
fixed